src/utils/ReactChildren.js

```diff
@@ -81,7 +81,7 @@ function mapSingleChildIntoContext(traverseContext, child, name, i) {
   var mapBookKeeping = traverseContext;
   var mapResult = mapBookKeeping.mapResult;
 
-  var keyUnique = !mapResult.hasOwnProperty(name);
+  var keyUnique = ('undefined' === typeof mapResult[name]);
```

Comment:
We have a transpiler that can turn `=== undefined` into `=== void 0` which is faster yet but ugly.